test/test_testing.py

```diff
@@ -4914,6 +4915,39 @@ def test_trigger_secret_expiration(self):
         with self.assertRaises(RuntimeError):
             harness.trigger_secret_removal('nosecret', 1)
 
+    def test_secret_permissions(self):
```

Comment:
this test makes my eyes bleed. Can we split it, parametrize it, or at least add a few blank newlines? :smiling_face_with_tear:

Reply:
:). I've split it. parameterisation would be nicest, but I find that a pain to do with unittest, so...